code/modules/assembly/signaler.dm

```diff
@@ -60,128 +60,134 @@
-			connected.Pulse(src)
-		else if(holder)
-			holder.process_activation(src, 1, 0)
-		else
-			..(radio)
-		return 1
+/obj/item/device/assembly/signaler/pulse(var/radio = 0)
+	if(src.connected && src.wires)
+		connected.Pulse(src)
+	else if(holder)
+		holder.process_activation(src, 1, 0)
+	else
+		..(radio)
+	return 1
 
 
-	receive_signal(datum/signal/signal)
-		if(!signal)
-			return 0
+/obj/item/device/assembly/signaler/receive_signal(datum/signal/signal)
+	if(!signal)
+		return 0
 
-		if(within_jamming_range(src))
-			return 0
+	if(within_jamming_range(src))
+		return 0
 
-		if(signal.encryption != code)
-			return 0
+	if(signal.encryption != code)
+		return 0
 
-		if(!(src.wires & WIRE_RADIO_RECEIVE))
-			return 0
+	if(!(src.wires & WIRE_RADIO_RECEIVE))
+		return 0
 
-		pulse(1)
+	pulse(1)
 
-		if(!holder)
-			for(var/mob/O in hearers(1, src.loc))
-				O.show_message(text("\icon[] *beep* *beep*", src), 3, "*beep* *beep*", 2)
-		return
+	if(!holder)
+		for(var/mob/O in hearers(1, src.loc))
+			O.show_message(text("\icon[] *beep* *beep*", src), 3, "*beep* *beep*", 2)
+	return
 
 
-	proc/set_frequency(new_frequency)
-		if(!frequency)
-			return
-		if(!SSradio)
-			sleep(20)
-		if(!SSradio)
-			return
-		SSradio.remove_object(src, frequency)
-		frequency = new_frequency
-		radio_connection = SSradio.add_object(src, frequency, RADIO_CHAT)
+/obj/item/device/assembly/signaler/proc/set_frequency(new_frequency)
+	if(!frequency)
 		return
-
-	process()
-		if(!deadman)
-			STOP_PROCESSING(SSprocessing, src)
-		var/mob/M = src.loc
-		if(!M || !ismob(M))
-			if(prob(5))
-				signal()
-			deadman = 0
-			STOP_PROCESSING(SSprocessing, src)
-		else if(prob(5))
-			M.visible_message("[M]'s finger twitches a bit over [src]'s signal button!")
+	if(!SSradio)
+		sleep(20)
+	if(!SSradio)
 		return
-
-	verb/deadman_it()
-		set src in usr
-		set name = "Threaten to push the button!"
-		set desc = "BOOOOM!"
+	SSradio.remove_object(src, frequency)
+	frequency = new_frequency
+	radio_connection = SSradio.add_object(src, frequency, RADIO_CHAT)
+	return
+
+/obj/item/device/assembly/signaler/process()
+	if(!deadman)
+		STOP_PROCESSING(SSprocessing, src)
+	var/mob/M = src.loc
+	if(!M || !ismob(M))
+		src.visible_message("<span class='warning'>[src]'s deadman switch is released !</span>")
+		signal()
+		deadman = 0
+		STOP_PROCESSING(SSprocessing, src)
+	else if(prob(20))
+		M.visible_message("[M]'s finger twitches a bit over [src]'s deadman switch!")
+	return
+
+/obj/item/device/assembly/signaler/verb/deadman_it()
+	set src in usr
+	set name = "Threaten to push the button!"
+	set desc = "BOOOOM!"
+	if(!deadman)
 		deadman = 1
 		START_PROCESSING(SSprocessing, src)
 		log_and_message_admins("is threatening to trigger a signaler deadman's switch")
-		usr.visible_message("<span class='warning'>[usr] moves their finger over [src]'s signal button...</span>")
+		usr.visible_message("<span class='warning'>[usr] activates [src]'s deadman switch</span>")

```

Comment:
Not a fan of this message. Too explicit in describing what's happening.

Reply:
What should be used instead ?

Reply:
Old message was fine imo. Should describe actions, not exactly what's happening.